### ISSUE TITLE
Empty type when `patternProperties` is used.

### ISSUE THREAD

**Issue by novafacing** [oxidecomputer/typify#635]
I'm generating types for the [tree-sitter](https://github.com/tree-sitter/tree-sitter) [schema](https://github.com/tree-sitter/tree-sitter/blob/master/cli/src/generate/grammar-schema.json). The `rules` property in that schema is:

```json
"rules": {
  "type": "object",
  "patternProperties": {
    "^[a-zA-Z_]\\w*$": {
      "$ref": "#/definitions/rule"
    }
  },
  "additionalProperties": false
},
```

This generates a type which is just an empty struct, instead of what it probably should be, which is a `HashMap<String, Rule>`.

I'm working on a fix, but wanted to create an issue for tracking.

Edit: The generated type is:
```rust
#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(deny_unknown_fields)]
pub struct TreeSitterGrammarSpecificationRules {}
impl From<&TreeSitterGrammarSpecificationRules> for TreeSitterGrammarSpecificationRules {
    fn from(value: &TreeSitterGrammarSpecificationRules) -> Self {
        value.clone()
    }
}
```

**Comment by novafacing** [comment]
This specific issue would be closed by #636 but only as a stopgap -- there are other places in JSON Schema where key validation is possible, and those would probably best be implemented as deserialize_with overrides against a regex.

**Comment by novafacing** [comment]
Re-implemented this correctly in #637 so that keys are actually validated against a regex.
